datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -69,6 +72,23 @@ pub struct InListExpr {
     expr: Arc<dyn PhysicalExpr>,
     list: Vec<Arc<dyn PhysicalExpr>>,
     negated: bool,
+    set: Option<InSet>,
+}
+
+/// InSet
+#[derive(Debug)]
+pub struct InSet {
+    set: HashSet<ScalarValue>,
```

Comment:
But this makes something for a future PR

Reply:
@Dandandan Thanks for your information❤️, Is there any specific reason why using ScalarValue is slower?

Reply:
I think it's because `ScalarValue` is an enum of a wrapper of value. So it would be overheads for both memory footprint and computation compared to HashSet of native data values.

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/2165

Reply:
In addition to higher memory usage and dispatching overhead there are two extra sources of overhead

* Having to convert all values from array items to `ScalarValue`
> * Hashing a `Scalarvalue` is slower than hashing a native type.